### PR TITLE
[DESign2-56] Add Link component 

### PR DESCRIPTION
**Pull request by levadadenys**
## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

### [DESign2-56] Add Link component

* implementation of Link component
* added storybook
* added tests
* added documentation

https://github.com/code-dot-org/code-dot-org/assets/22244040/89ca61c6-0868-4ee1-a6f1-f91451d3ff2d

## Links

- jira ticket: [Design2-56](https://codedotorg.atlassian.net/browse/DESIGN2-56)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/link/Link.tsx [hunk 1]
```diff
@@ -5,27 +5,71 @@ import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
...
+  /** Link destination */
+  href: string;
+  /** Is the link disabled? */
+  disabled?: boolean;
+  /** Callback for click event */
+  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
   /** Size of link */
   size?: ComponentSizeXSToL;
+  /** Color/type of link */
+  color?: 'primary' | 'secondary';
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/link/Link.tsx]: `color` being either "primary" or "secondary" reads a little confusing to me. Would `type` make more sense?

File: apps/test/unit/componentLibrary/LinkTest.jsx [hunk 2]
```diff
@@ -1,246 +1,85 @@
...
+import userEvent from '@testing-library/user-event';
+import sinon from 'sinon';
+
+import {expect} from '../../util/reconfiguredChai';
+
+import Link from '@cdo/apps/componentLibrary/link';
+
+describe('Design System - Link', () => {
+  it('Link - renders with correct text', () => {
+    render(<Link href="https://studio.code.org/home">Home</Link>);
```
Comment (sanchitmalhotra126) [inline apps/test/unit/componentLibrary/LinkTest.jsx:12]: nit: constant for `"https://studio.code.org/home"` just to ensure the test is referencing the same value? also can be reused across multiple tests

File: apps/test/unit/componentLibrary/LinkTest.jsx [hunk 3]
```diff
@@ -1,246 +1,85 @@
...
+  });
+
+  it('Link - calls onClick', async () => {
+    const user = userEvent.setup();
+    const spyOnClick = sinon.spy();
+
+    const {rerender} = render(
+      <Link href="/" onClick={spyOnClick}>
+        Home
+      </Link>
```
Comment (sanchitmalhotra126) [inline apps/test/unit/componentLibrary/LinkTest.jsx]: nit: save this component to a local to reuse in `rerender`?

**Review by sanchitmalhotra126 — APPROVED** [review]
Looks good! Just a few minor comments

File: apps/src/componentLibrary/link/Link.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/link/Link.tsx]: Makes sense, thanks for the catch!

File: apps/test/unit/componentLibrary/LinkTest.jsx (hunk 3 above)
Comment (levadadenys) [reply apps/test/unit/componentLibrary/LinkTest.jsx]: Not sure I understand, can you explain or show example maybe please?

Comment (sanchitmalhotra126) [reply apps/test/unit/componentLibrary/LinkTest.jsx]: Oh yep just meant
```
const component =  (
  <Link href="/" onClick={spyOnClick}>
     Home
  </Link>
);

const {rerender} = render(component);
...
rerender(component);
```
Unless they need to be separate components?

Comment (levadadenys) [reply apps/test/unit/componentLibrary/LinkTest.jsx]: Oh yeah, nice idea, thanks for spotting this.
I've checked, everything is working correctly. Unless we gonna need to manually update the component's props between `rerender`  - we can use `const component = ...` approach

**Pull request merged**